clients/drcachesim/tests/cache_replacement_policy_unit_test.cpp

```diff
@@ -0,0 +1,133 @@
+/* **********************************************************
+ * Copyright (c) 2016-2022 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+// Unit tests for cache replacement policies
+#include <iostream>
+#include <assert.h>
+#include "cache_replacement_policy_unit_test.h"
+#include "simulator/cache_lru.h"
+
+class cache_lru_test_t : public cache_lru_t {
+public:
+    void
+    initialize_cache(int associativity, int line_size, int total_size)
+    {
+        caching_device_stats_t *stats = new cache_stats_t(line_size, "", true);
+        if (!init(associativity, line_size, total_size, nullptr, stats, nullptr)) {
+            std::cerr << "LRU cache failed to initialize"
+                      << "\n";
+            exit(1);
+        }
+    }
+
+    void
+    unit_test_replace_which_way()
+    {
+        // Create and initialize a 4-way set associative cache with line size of 32 and
+        // total size of 256 bytes.
```

Comment:
suggest: Instead of this separate comment which duplicates the numbers (and thus can get out of sync and require changing in two places), use parameter name comments:
```C
initialize_cache(/*associativity=*/4, /*line_size=*/32, /*total_size=*/256);
```